### DISCUSSION TITLE
How to enforce stricter `Any` type checking in Pyright strict mode?

### DISCUSSION THREAD

**Discussion by rafrafek** [microsoft/pyright#9611] [Q&A] [answered]
Hello,
I'm using Pyright in strict mode, and I need the `Any` type to be validated. I want the type checker to prevent me from using `Any` where a function specifically requires a string, but allow it for functions that can handle a parameter of any type. I'm not sure whether `Any` means "any type" or "ignore type" in Python. Should I continue using `Any`, or should I use a different type annotation, like `object`, to achieve that?

```python
from typing import Any


def accept_any(foo: Any) -> None:
    print(type(foo))  # Should be OK, but it's not OK


def accept_str(bar: str) -> None:
    print(type(bar))


def call_the_functions_above(foo: Any) -> None:
    accept_any(foo)  # Should be OK, and it's OK

    accept_str(foo)  # Should not be OK, but it's OK
```

**Comment by erictraut** [comment, marked as answer]
`Any` has a very specific meaning in the Python typing spec, and pyright conforms to the spec. For more details, refer to [this section of the spec](https://typing.readthedocs.io/en/latest/spec/concepts.html#gradual-types).

If you want your code to be fully type checked, you should avoid using `Any` wherever possible. When you use `Any`, you're effectively telling the type checker "I don't know what type this is, so assume that it's any type that could possibly satisfy type checking constraints".

**Reply by rafrafek** [reply]
Thank you for the clarification! That makes sense. I'll use `object` instead of `Any` for stricter type validation.
